### PR TITLE
DEV: puts debugging information when CI encounters resolution errors

### PR DESCRIPTION
**Pull request by tgxworld**
We have been seeing flaky socket resolution errors on CI and need more
debugging information to figure out why

### CODE REVIEW

File: spec/rails_helper.rb [hunk 1]
```diff
@@ -518,6 +518,22 @@ class SpecTimeoutError < StandardError
         sleep 0.01 while !mutex.synchronize { is_waiting }
       end
     end
+
+    config.around do |example|
```
Comment (tgxworld) [inline spec/rails_helper.rb:522]: This block is only enabled on `ENV["CI"]`

**Review by martin-brennan — APPROVED** [review]

File: spec/rails_helper.rb [hunk 2]
```diff
@@ -518,6 +518,22 @@ class SpecTimeoutError < StandardError
...
+      example.run
+
+      if example.exception.is_a?(Socket::ResolutionError)
+        require "socket"
```
Comment (Drenmi) [inline spec/rails_helper.rb:526]: Not a blocker, but curious: shouldn't `Socket` be available if we're referencing `Socket::ResolutionError` in the conditional?

**Review by Drenmi — APPROVED** [review]
🧦

File: spec/rails_helper.rb (hunk 2 above)
Comment (tgxworld) [reply spec/rails_helper.rb:526]: Good point :+1

**Pull request merged**